include/solvers/heat_transfer.h

```diff
@@ -0,0 +1,273 @@
+/* ---------------------------------------------------------------------
+ *
+ * Copyright (C) 2019 - 2020 by the Lethe authors
+ *
+ * This file is part of the Lethe library
+ *
+ * The Lethe library is free software; you can use it, redistribute
+ * it, and/or modify it under the terms of the GNU Lesser General
+ * Public License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * The full text of the license can be found in the file LICENSE at
+ * the top level of the Lethe distribution.
+ *
+ * ---------------------------------------------------------------------
+
+ *
+ * Implementation of heat transfer as an auxiliary physics.
+ * This heat equation is weakly coupled to the velocity field.
+ * Equation solved:
+ * rho * Cp * (dT/dt + u.gradT) = k div(gradT) + nu/rho * (gradu : gradu)
+ *
+ * Author: Bruno Blais, Polytechnique Montreal, 2020-
+ */
+
+#ifndef lethe_heat_transfer_h
+#define lethe_heat_transfer_h
+
+#include <deal.II/base/convergence_table.h>
+
+#include <deal.II/distributed/solution_transfer.h>
+#include <deal.II/distributed/tria_base.h>
+
+#include <deal.II/fe/fe_q.h>
+
+#include <deal.II/lac/trilinos_sparse_matrix.h>
+#include <deal.II/lac/trilinos_vector.h>
+
+#include <core/simulation_control.h>
+#include <solvers/auxiliary_physics.h>
+#include <solvers/multiphysics_interface.h>
+
+
+template <int dim>
+class HeatTransfer : public AuxiliaryPhysics<dim, TrilinosWrappers::MPI::Vector>
+{
+public:
+  HeatTransfer<dim>(MultiphysicsInterface<dim> *     multiphysics_interface,
+                    const SimulationParameters<dim> &p_simulation_parameters,
+                    std::shared_ptr<parallel::DistributedTriangulationBase<dim>>
+                                                       p_triangulation,
+                    std::shared_ptr<SimulationControl> p_simulation_control)
+    : AuxiliaryPhysics<dim, TrilinosWrappers::MPI::Vector>(
+        p_simulation_parameters.non_linear_solver)
+    , multiphysics(multiphysics_interface)
+    , simulation_parameters(p_simulation_parameters)
+    , triangulation(p_triangulation)
+    , simulation_control(p_simulation_control)
+    , dof_handler(*triangulation)
+    , fe(simulation_parameters.fem_parameters.temperature_order)
+    , solution_transfer(dof_handler)
+    , solution_transfer_m1(dof_handler)
+    , solution_transfer_m2(dof_handler)
+    , solution_transfer_m3(dof_handler)
+  {}
+
+  /**
+   * @brief Call for the assembly of the matrix and the right-hand side.
+   *
+   * @param time_stepping_method Time-Stepping method with which the assembly is called
+   */
+  virtual void
+  assemble_matrix_and_rhs(
+    const Parameters::SimulationControl::TimeSteppingMethod
+      time_stepping_method) override;
```

Comment:
Override does more than that. It will give you a compilation error if you are not overloading a function. It's more there as a safeguard than anything.